core/transaction_pool.go

```diff
@@ -71,7 +71,12 @@ type TxPool struct {
 	queue   map[common.Address]map[common.Hash]*types.Transaction
 }
 
-func NewTxPool(eventMux *event.TypeMux, currentStateFn stateFn, gasLimitFn func() *big.Int) *TxPool {
+func NewTxPool(eventMux *event.TypeMux, currentStateFn stateFn, gasLimitFn func() *big.Int) (*TxPool, error) {
+	currentState, err := currentStateFn()
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Looking at it another way: the txpool calls `currentStateFn` many times throughout its lifecycle,
any of these calls can fail with a missing root error. Why should the first call prevent creating the pool?